frontend/src/commit-panel/commit/Form.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+
+interface FormProps {
+  commitMessage: string;
+  onCommitMessageChange(e: React.FormEvent): void;
+  onSubmit(e: React.FormEvent): void;
+  onCancelCommitClick(e: React.MouseEvent): void;
+}
+
+const Form: React.StatelessComponent<FormProps> = (
+  { commitMessage, onCommitMessageChange, onSubmit, onCancelCommitClick }) => (
```

Comment:
I agree that this format style is not a great one, but I don't believe that destructuring in the function body is the way to go. Recently I've used this style for such a stateless component:

```jsx
const Form: React.StatelessComponent<FormProps> = ({
  commitMessage, onCommitMessageChange, onSubmit, onCancelCommitClick,
}) => (
  // ...
);
```

What do you think about it?